subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpResourceAccessor.java

```diff
@@ -61,28 +59,20 @@ public HttpResponseResource openResource(final URI uri) {
      * Same as #getResource except that it always gives access to the response body,
      * irrespective of the returned HTTP status code. Never returns {@code null}.
      */
-    public HttpResponseResource getRawResource(final URI uri) {
+    public HttpResponseResource getRawResource(final URI uri, boolean revalidate) {
         abortOpenResources();
         String location = uri.toString();
         LOGGER.debug("Constructing external resource: {}", location);
-
-        HttpRequestBase request = new HttpGet(uri);
-        HttpResponse response;
-        try {
-            response = http.performHttpRequest(request);
-        } catch (IOException e) {
-            throw new HttpRequestException(String.format("Could not %s '%s'.", request.getMethod(), request.getURI()), e);
-        }
-
+        HttpResponse response = http.performGet(location, revalidate);
```

Comment:
Actually, looking at this closer, it's not the HttpRequestException that's important, its the fact that response is  now null instead of being a 404 response object (previous behavior).  This should probably be `performRawGet()` in order to maintain the previous functionality.